vpnmanager/vpnservicemanager/src/main/java/com/kape/vpnservicemanager/data/externals/Service.kt

```diff
@@ -41,18 +41,34 @@ internal class Service :
     private lateinit var protocol: IProtocol
     private lateinit var subnet: ISubnet
     private lateinit var cache: ICache
+    private lateinit var onServiceRevoked: () -> Unit
 
     // region VpnService
-    override fun onBind(intent: Intent?): IBinder {
-        return binder
+    override fun onBind(intent: Intent?): IBinder? {
+        return if (intent?.action == SERVICE_INTERFACE) {
+            // Passing in a custom binder for SERVICE_INTERFACE action would prevent the #onRevoke() from being called.
+            super.onBind(intent)
```

Comment:
so, is this is now the default behavior in that case? The concern that I have is that we might be operating on different binders. If you follow `ServiceBinder` you will see that it's used to get the service on `onServiceConnected` -> `cache.setService(binder.getService())`. and I'm not sure if there would be any undefined behaviours for returning a Binder here that's not the one we are using later on for getting the service. Have you tried that particular scenario?

Reply:
I actually picked the suggestion from this Slack thread: https://stackoverflow.com/questions/15513144/how-to-add-additional-handlings-to-vpn-disconnection-in-android

I also had the same doubts about passing in a different Binder and then trying to cast it to `ServiceBinder` in `onServiceConnected` method. But I tested starting and stopping a VPN connection multiple times and it never crashed. I also testing stopping it from the Android settings menu just to be sure. I also forced it to stop by using another VPN app to trigger a connection and it worked flawlessly in all the above scenarios.

Reply:
Perfect. Thank you!